integration_tests/suite_test.go

```diff
@@ -94,7 +94,14 @@ var _ = BeforeSuite(func() {
 		EnableRemoteSecrets: true,
 	}
 
-	Expect(controllers.SetupAllReconcilers(mgr, ITest.OperatorConfiguration, storage)).To(Succeed())
+	ITest.ClientFactory = TestClientFactory{
+		GetClientImpl: func(_ context.Context, _ string, _ *api.RemoteSecretTarget, _ *api.TargetStatus) (client.Client, error) {
+			// effectively, this switches off any support for auth or deployment to remote clusters in the integration tests..
```

Comment:
I would leave this for the e2e tests personally, because the setup would be quite involved and we would still not be able to test this fully in integration tests because testenv doesn't support auth/z properly.